operators/src/processing/interpolation/mod.rs

```diff
@@ -0,0 +1,824 @@
+            .spatial_partition()
+            .upper_left();
+
+        let geo_transform = GeoTransform::new(
+            origin_coordinate,
+            query_rect.spatial_resolution.x,
+            -query_rect.spatial_resolution.y,
+        );
+
+        let bbox = tiling.tile_grid_box(query_rect.spatial_bounds);
+
+        let shape = [
+            bbox.axis_size_y() * tiling.tile_size_in_pixels.axis_size_y(),
+            bbox.axis_size_x() * tiling.tile_size_in_pixels.axis_size_x(),
+        ];
+
+        // create a non-aligned (w.r.t. the tiling specification) grid by setting the origin to the top-left of the tile and the tile-index to [0, 0]
+        let grid = Grid2D::new(
+            shape.into(),
+            vec![no_data_value; shape[0] * shape[1]],
+            Some(no_data_value),
+        )
+        .expect("grid creation must not fail");
+
+        let input_tile = RasterTile2D::new(
+            query_rect.time_interval,
+            [0, 0].into(),
+            geo_transform,
+            GridOrEmpty::Grid(grid),
+        );
+
+        let output_tile = RasterTile2D::new_with_tile_info(
+            query_rect.time_interval,
+            tile_info,
+            GridOrEmpty::Empty(EmptyGrid::new(
+                tiling_specification.tile_size_in_pixels,
+                no_data_value,
+            )),
+        );
+
+        InterpolationAccu::new(input_tile, output_tile, pool)
+    })
+    .map_err(From::from)
+}
+
+pub fn fold_future<T, I>(
+    accu: InterpolationAccu<T, I>,
+    tile: RasterTile2D<T>,
+) -> impl Future<Output = Result<InterpolationAccu<T, I>>>
+where
+    T: Pixel,
+    I: InterpolationAlgorithm<T>,
+{
+    crate::util::spawn_blocking(|| fold_impl(accu, tile)).then(|x| async move {
+        match x {
+            Ok(r) => r,
+            Err(e) => Err(e.into()),
+        }
+    })
+}
+
+pub fn fold_impl<T, I>(
+    mut accu: InterpolationAccu<T, I>,
+    tile: RasterTile2D<T>,
+) -> Result<InterpolationAccu<T, I>>
+where
+    T: Pixel,
+    I: InterpolationAlgorithm<T>,
+{
+    // get the time now because it is not known when the accu was created
+    accu.output_tile.time = tile.time;
+
+    // copy all input tiles into the accu to have all data for interpolation
+    let mut accu_input_tile = accu.input_tile.into_materialized_tile();
+    accu_input_tile.blit(tile)?;
+
+    Ok(InterpolationAccu::new(
+        accu_input_tile.into(),
+        accu.output_tile,
+        accu.pool,
+    ))
+}
+
+#[async_trait]
+pub trait InterpolationAlgorithm<P: Pixel>: Send + Sync + Clone + 'static {
+    /// interpolate the given input tile into the output tile
+    /// the output must be fully contained in the input tile and have an additional row and column in order
+    /// to have all the required neighbor pixels.
+    /// Also the output must have a finer resolution than the input
+    fn interpolate(input: &RasterTile2D<P>, output: &mut MaterializedRasterTile2D<P>)
+        -> Result<()>;
+}
```

Comment:
it should be possible to add it, but not "simply" because the bicubic interpolation needs 16 neighbor pixels instead of four, so we would need to make the creation of the accumulator dependent on the selected interpolation algorithm and make the query rectangle even larger for bicubic.